addons/point_of_sale/static/src/app/models/related_models.js

```diff
@@ -474,652 +463,666 @@ export function createRelatedModels(modelDefs, modelClasses = {}, opts = {}) {
+                } else {
+                    result[name] = record[name] !== undefined ? record[name] : false;
+                }
+            }
+            return result;
+        }
+        // aliases
+        getAllBy() {
+            return this.readAllBy(...arguments);
+        }
+        getAll() {
+            return this.readAll(...arguments);
+        }
+        getBy() {
+            return this.readBy(...arguments);
+        }
+        get() {
+            return this.read(...arguments);
+        }
+        getFirst() {
+            return this.readFirst(...arguments);
+        }
+        // array prototype
+        map(fn) {
+            return this.orderedRecords.map(fn);
+        }
+        reduce(fn, initialValue) {
+            return this.orderedRecords.reduce(fn, initialValue);
+        }
+        flatMap(fn) {
+            return this.orderedRecords.flatMap(fn);
+        }
+        forEach(fn) {
+            return this.orderedRecords.forEach(fn);
+        }
+        some(fn) {
+            return this.orderedRecords.some(fn);
         }
+        every(fn) {
+            return this.orderedRecords.every(fn);
+        }
+        find(fn) {
+            return this.orderedRecords.find(fn);
+        }
+        filter(fn) {
+            return this.orderedRecords.filter(fn);
+        }
+        sort(fn) {
+            return this.orderedRecords.sort(fn);
+        }
+        indexOf(record) {
+            return this.orderedRecords.indexOf(record);
+        }
+        get length() {
+            return this.records[this.name].size;
+        }
+        // External callbacks
+        addEventListener(event, callback) {
+            if (!AVAILABLE_EVENT.includes(event)) {
+                throw new Error(`Event '${event}' is not available`);
+            }
+
+            if (!(event in callbacks[this.name])) {
+                callbacks[this.name][event] = [];
+            }
 
-        if (!baseData[model][id]) {
-            baseData[model][id] = vals;
+            callbacks[this.name][event].push(callback);
         }
+        triggerEvents(event, data) {
+            if (
+                !(event in callbacks[this.name]) ||
+                callbacks[this.name][event].length === 0 ||
+                !data
+            ) {
+                return;
+            }
 
-        record._raw = baseData[model][id];
-        crud.records[model].set(id, record);
+            for (const callback of callbacks[this.name][event]) {
+                callback(data);
+            }
+        }
 
-        const fields = getFields(model);
-        for (const name in fields) {
-            if (name === "id") {
-                continue;
+        _create(vals, ignoreRelations = false, fromSerialized = false, delayedSetup = false) {
+            if (!("id" in vals)) {
+                vals["id"] = getLocalId(this.name);
             }
 
-            const field = fields[name];
+            const record = toRaw(this.instantiate());
```

Comment:
toRaw?

Reply:
Correct me if I'm wrong but I believe that before this commit, the created record is not yet reactive before going thru the for-loop. Then, at the end, we return the reactive version.